runtime-modules/argo-bridge/src/types.rs

```diff
@@ -0,0 +1,61 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{storage::bounded_vec::BoundedVec, traits::ConstU32};
+use scale_info::TypeInfo;
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+use sp_std::vec::Vec;
+
+// Balance type alias
+pub type BalanceOf<T> = <T as balances::Config>::Balance;
+
+pub type ChainId = u32;
+pub type TransferId = u64;
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, TypeInfo, MaxEncodedLen)]
+pub struct BridgeConstraints<AccountId, Balance, BlockNumber> {
+    pub operator_account: Option<AccountId>,
+    pub pauser_accounts: Option<Vec<AccountId>>,
+    pub bridging_fee: Option<Balance>,
+    pub thawn_duration: Option<BlockNumber>,
+    pub remote_chains: Option<BoundedVec<ChainId, ConstU32<10>>>,
+}
+
+pub type BridgeConstraintsOf<T> = BridgeConstraints<
+    <T as frame_system::Config>::AccountId,
+    BalanceOf<T>,
+    <T as frame_system::Config>::BlockNumber,
+>;
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, Debug, TypeInfo, MaxEncodedLen)]
+pub struct RemoteAccount {
+    // Ethereum addresses only need 20 bytes but we use 32 bytes to enable compatibility with other chains
+    // When using Ethereum addresses, the last 12 bytes should be zero
```

Comment:
```suggestion
    // When using EVM-based addresses, the last 12 bytes should be zero
```

Reply:
The most significant 12 bytes should be 0, this padding method is consistent with the way Solidity handles addresses and other data types, ensuring compatibility and consistency across the Ethereum ecosystem.
So the first 12 bytes should be 0 and not the last 12

Reply:
cc: @kdembler this decision might have consequence on how you parse events fields on the argo squid

Reply:
Yeah that's what I meant actually, my mistake writing spec. I suggest
```suggestion
    // When using EVM addresses, the 12 most-significant bytes should be zero
```
Squid already uses this approach